src/stream/src/task/stream_manager.rs

```diff
@@ -732,8 +733,18 @@ impl StreamActorManagerState {
     fn update_actors(&mut self, actors: Vec<stream_plan::StreamActor>) -> StreamResult<()> {
         for actor in actors {
             let actor_id = actor.get_actor_id();
+            let fragment_id = actor.get_fragment_id();
+            let expr_context = actor.get_expr_context();
+            let mut new_actor = actor.clone();
+            if let Ok(value) = expr_context {
+                let _ = new_actor.expr_context.insert(ExprContext {
+                    time_zone: value.time_zone.clone(),
+                    fragment_id,
+                });
+            }
+
             self.actors
-                .try_insert(actor_id, actor)
+                .try_insert(actor_id, new_actor)
```

Comment:
Not sure whether here is a good place to fill the `fragment_id` info into `expr_context`.